src/seedu/addressbook/AddressBook.java

```diff
@@ -36,1138 +36,1140 @@
+	 * used by the internal String[] storage format.
+	 * For example, a person's name is stored as the 0th element in the array.
+	 */
+	private static final int PERSON_DATA_INDEX_NAME = 0;
+	private static final int PERSON_DATA_INDEX_PHONE = 1;
+	private static final int PERSON_DATA_INDEX_EMAIL = 2;
+	private static final int NUM_ARGUMENTS = 3;
+
+	/**
+	 * The number of data elements for a single person.
+	 */
+	private static final int PERSON_DATA_COUNT = 3;
+
+	/**
+	 * Offset required to convert between 1-indexing and 0-indexing.COMMAND_
+	 */
+	private static final int DISPLAYED_INDEX_OFFSET = 1;
+
+	/**
+	 * If the first non-whitespace character in a user's input line is this, that line will be ignored.
+	 */
+	private static final char INPUT_COMMENT_MARKER = '#';
+
+	/*
+	 * This variable is declared for the whole class (instead of declaring it
+	 * inside the readUserCommand() method to facilitate automated testing using
+	 * the I/O redirection technique. If not, only the first line of the input
+	 * text file will be processed.
+	 */
+	private static final Scanner SCANNER = new Scanner(System.in);
+
+	/*
+	 * NOTE : =============================================================================================
+	 * Note that the type of the variable below can also be declared as List<String[]>, as follows:
+	 *    private static final List<String[]> ALL_PERSONS = new ArrayList<>()
+	 * That is because List is an interface implemented by the ArrayList class.
+	 * In this code we use ArrayList instead because we wanted to to stay away from advanced concepts
+	 * such as interface inheritance.
+	 * ====================================================================================================
+	 */
+
+	/**
+	 * List of all persons in the address book.
+	 */
+	private static final ArrayList<String[]> ALL_PERSONS = new ArrayList<>();
+
+	/**
+	 * Stores the most recent list of persons shown to the user as a result of a user command.
+	 * This is a subset of the full list. Deleting persons in the pull list does not delete
+	 * those persons from this list.
+	 */
+	private static ArrayList<String[]> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
+
+	/**
+	 * The path to the file used for storing person data.
+	 */
+	private static String storageFilePath;
+
+	/*
+	 * NOTE : =============================================================
+	 * Notice how this method solves the whole problem at a very high level.
+	 * We can understand the high-level logic of the program by reading this
+	 * method alone.
+	 * If the reader wants a deeper understanding of the solution, she can go
+	 * to the next level of abstraction by reading the methods that are
+	 * referenced by the high-level method below.
+	 * ====================================================================
+	 */
+
+	public static void main(String[] args) {
+		showWelcomeMessage();
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.

This comment is applicable to the rest of the code as well